src/field/captcha/third_party_captcha.jsx

```diff
@@ -169,23 +182,40 @@ export class ThirdPartyCaptcha extends React.Component {
         if (this.state.retryCount < MAX_RETRY) {
           removeScript(scriptUrl);
           loadScript(scriptUrl, attributes);
-          this.setState((prevState) => ({
+          this.setState(prevState => ({
             retryCount: prevState.retryCount + 1
           }));
           return;
         }
         removeScript(scriptUrl);
         this.changeHandler('BYPASS_CAPTCHA');
       };
-      window[callbackName] = (arkose) => {
+      window[callbackName] = arkose => {
         callback(arkose);
       };
+    } else if (provider === AUTH0_V2_CAPTCHA_PROVIDER) {
+      attributes['error-callback'] = () => {
+        if (this.state.retryCount < MAX_RETRY) {
+          removeScript(scriptUrl);
+          loadScript(scriptUrl, attributes);
+          this.setState(prevState => ({
+            retryCount: prevState.retryCount + 1
+          }));
+          return;
+        }
+        removeScript(scriptUrl);
+        this.changeHandler('BYPASS_CAPTCHA');
+      };
+      window[callbackName] = () => {
+        delete window[callbackName];
+        callback();
```

Comment:
Can we reuse the code we already have for this, as it seems to be identical apart from two things:

- `onerror` vs `error-callback`
- arkose is not doing `delete window[callbackName]`. Why is that different here?

Reply:
The naming comes directly from the APIs so we can't align them directly. I didn't work on the arkose changes but aside from skipping the delete they also send in the arkose script and I know they handle the callback a bit differently further up in the tree. I will look into it to get some more insight and get back to you!

Reply:
I didnt mean to align the names. I just mean that it may be possible to reuse the same code. Along the lines of:

```js
failOpen(attributes, callbackName, errorCallbackName, callback) {
  attributes[errorCallbackName] = () => {
        if (this.state.retryCount < MAX_RETRY) {
          removeScript(scriptUrl);
          loadScript(scriptUrl, attributes);
          this.setState(prevState => ({
            retryCount: prevState.retryCount + 1
          }));
          return;
        }
        removeScript(scriptUrl);
        this.changeHandler('BYPASS_CAPTCHA');
      };
    window[callbackName] = callback;
}

if (provider === ARKOSE_PROVIDER || provider === AUTH0_V2_CAPTCHA_PROVIDER) {
  const errorCallbackName = provider === ARKOSE_PROVIDER ? 'onerror' : 'error-callback';
  failOpen(attributes, callbackName, errorCallbackName, callback);
}
```

Anyway, not saying you have to make this change. But it felt like it was doing the exact same thing so maybe worth considering but can approve if you believe it makes no sense.

Reply:
The above doesnt do the `delete window[callbackName]`, so maybe it needs to be added. But anyway, I just wanted to show what I meant.

Reply:
See my other [comment](https://github.com/auth0/lock/pull/2507#discussion_r1433587240) but we can definitely combine the Arkose & Auth0 V2 error handling here. The attributes here refers to the attributes of the script tag and we're trying to add error handling in the case the script fails to load to the DOM and this is not the same as `error-callback` param from the provider API which is after the script is loaded and when there are provider API related errors.